src/Symfony/Component/Lock/Store/SemaphoreStore.php

```diff
@@ -33,6 +33,10 @@ class SemaphoreStore implements StoreInterface
      */
     public static function isSupported($blocking = null)
     {
+        if (null !== $blocking) {
+            @trigger_error(sprintf('The "$blocking" argument of the "%s" is deprecated since 3.4 and will be removed in 4.0.', __METHOD__), E_USER_DEPRECATED);
```

Comment:
Internal call should not trigger deprecation (this argument is used by component console)

Reply:
But how know if call is internal or not?

Reply:
By passing `false` as second (extra, implicit) argument in each internal call, then triggering only if `1 === func_num_args() || func_get_arg(1)`. But wait, @jderusse do we really need to expose this argument in 3.4? Would it be enough to not add it and consider it exists only for internal calls?

Reply:
let's make it final and move on :)

Reply:
@nicolas-grekas doesn't https://github.com/symfony/symfony/pull/24053#discussion_r136384378 still apply when calling the method?

Reply:
nothing will break, so we don't care

Reply:
ok then :)

Reply:
I would just mark this method as `@internal`

Reply:
`@internal` used.